service/history/mutableStateDecisionTaskManager.go

```diff
@@ -0,0 +1,726 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package history
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	workflow "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/errors"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/persistence"
+)
+
+type (
+	mutableStateDecisionTaskManager interface {
+		ReplicateDecisionTaskScheduledEvent(
```

Comment:
could arrange the methods like
```
AddDecisionTaskScheduledEvent
ReplicateDecisionTaskScheduledEvent

AddDecisionTaskStartedEvent
ReplicateDecisionTaskStartedEvent
...
```

Reply:
There is not a clean mapping between the two for example ```AddDecisionTaskScheduledEventAsHeartbeat``` has not equal replicate method. So I think this grouping will look confusing.

Reply:
AddDecisionTaskScheduledEventAsHeartbeat has no counter part, that is true,
but for readability, we should group related logics in on section.

Reply:
The logics are grouped in related ways. We have all the Replicate methods together. We have all the Add*Events together etc...

Reply:
Resolved offline